src/main/java/org/rumbledb/runtime/functions/sequences/general/TreatAsClosure.java

```diff
@@ -4,36 +4,38 @@
 import org.rumbledb.api.Item;
 import org.rumbledb.exceptions.ExceptionMetadata;
 import org.rumbledb.exceptions.TreatException;
+import org.rumbledb.exceptions.UnexpectedTypeException;
 import org.rumbledb.types.SequenceType;
 
 public class TreatAsClosure implements Function<Item, Boolean> {
     private SequenceType sequenceType;
     private ExceptionMetadata metadata;
+    private boolean shouldThrowTreatException;
     private static final long serialVersionUID = 1L;
 
-    public TreatAsClosure(SequenceType sequenceType, ExceptionMetadata metadata) {
+    public TreatAsClosure(SequenceType sequenceType, boolean shouldThrowTreatException, ExceptionMetadata metadata) {
         this.sequenceType = sequenceType;
+        this.shouldThrowTreatException = shouldThrowTreatException;
         this.metadata = metadata;
     }
 
     @Override
     public Boolean call(Item input) throws Exception {
         if (!input.isTypeOf(this.sequenceType.getItemType())) {
-            throw new TreatException(
-                    " "
-                        + input.getDynamicType().toString()
-                        + " cannot be treated as type "
-                        + this.sequenceType.getItemType().toString()
-                        + this.sequenceType.getArity().getSymbol(),
-                    this.metadata
-            );
+            String message = input.getDynamicType().toString()
+                + " cannot be treated as type "
+                + this.sequenceType.getItemType().toString()
+                + this.sequenceType.getArity().getSymbol();
+            throw this.shouldThrowTreatException
+                ? new TreatException(message, this.metadata)
+                : new UnexpectedTypeException(message, this.metadata);
         }
         if (this.sequenceType.isEmptySequence()) {
-            throw new TreatException(
-                    input.getDynamicType().toString()
-                        + " cannot be treated as type empty-sequence()",
-                    this.metadata
-            );
+            String message = input.getDynamicType().toString()
+                + " cannot be treated as type empty-sequence()";
+            throw this.shouldThrowTreatException
+                ? new TreatException(message, this.metadata)
+                : new UnexpectedTypeException(message, this.metadata);
```

Comment:
I just wanted to ensure that we are happy with using the same error message for the treat exception and unexpected type exception. I feel they provide sufficient clarity in their current state. But I still wanted to share Zorba's error messages to take another look at this decision:

Zorba equivalents:
Query:
`1 + "A"`  
Error: 
`(no URI):1,2: type error [err:XPTY0004]: arithmetic operation not defined between types "xs:integer" and "xs:string"`

Query:
`"string" treat as xs:int`  
Error"
`(no URI):1,2: dynamic error [err:XPDY0050]: "xs:string" cannot be treated as type xs:int`

Reply:
DONE, good comment!